Source/JavaScriptCore/runtime/VM.cpp

```diff
@@ -475,6 +475,36 @@ VM::~VM()
 #endif
 }
 
+bool VM::validValueProfileDiagAnalysis(JSValue value)
+{
+    return value && value.isCell() && !value.isString();
+}
```

Comment:
Let's put this in VMInlines.h and make it an inline function.  Let's also rename it to something like `needsValueProfileCorruptionCheck`.